rfc-119-merge-mainstream-browse-topics-and-taxons.md

```diff
@@ -0,0 +1,183 @@
+
+Mainstream browse pages, specialist topics and taxons themselves are represented as content items. Each type can contain links to other items of the same type and usually do so building a hierarchical tree structure. For example, the "Child benefit" and "Universal Credit" specialist topics are linked and represented as children of the "Benefits" specialist topic. For each type there is one "root" item that represents the start of the hierarchy. There is an exception for taxons, where there are two hierarchies with two distinct roots and sets of items - [topic taxonomy][topic-taxonomy-docs] and [world taxonomy][world-taxonomy-docs]. For purposes of this RFC ignore the world taxonomy.
+
+
+## Problem
+
+We have to maintain three separate types of categorisation for content items - mainstream browse pages, specialist topics and taxons. This means the categories themselves and the content assigned to them may be different between each type. For example, the "Education and learning" mainstream browse page is the logical equivalent of the "Education, training and skills" taxon and should be the same, however they differ in name and in the set of content items they represent. There are also categories that only exist in some types and not others, for example the "Setting up a business" mainstream browse page has no equivalent taxon.
+
+Examples of equivalent content items:
+
+Mainstream browse page: https://www.gov.uk/api/content/browse/benefits/universal-credit
+
+Specialist topic: https://www.gov.uk/api/content/topic/benefits-credits/universal-credit
+
+Taxon: https://www.gov.uk/api/content/welfare/universal-credit
+
+There is currently ongoing work to ensure mainstream browse and specialist topics have equivalents in the taxonomy, i.e. mainstream browse and specialist topics become a subset of taxons. This is being done by Content Designers who are liaising with departments to come to agreement.
+
+Once this work is complete we want to maintain alignment and prevent divergence from taxons. We would have to put constraints in collections publisher and content tagger (or any future apps) to ensure that mainstream browse and specialist topics continue to be subsets of taxons. This would involve adding complex logic to the publishing system to deal with addition, changing or removal of these content items.
+
+## Proposal
+
+We propose merging the attributes of mainstream browse and specialist topic content items into **existing** taxon content items with the same `title`. (For clarity we aren't proposing the creation of new taxons or taxon tree.)
```

Comment:
I'm wondering how this would work for publishing apps tagging to these things. From other comments, I understand that mainstream browse and specialist topics will be a subset of the existing taxonomy. Once the merge has happened, how will we scope the total set of taxons so that a publisher can't accidentally tag to one that's not in the mainstream browse hierarchy, for example? (To be clear: this is where we're specifically asking them to "tag to mainstream browse categories" in the UI, so there's an expectation that the document will appear in certain places.)

Reply:
I can think of two options here:
1. Hide topics from publishing apps that don't correspond to user-facing topic pages
2. Allow tagging that's more granular that our user-facing topics show, but roll up this tagging to first user-facing level.
e.g. If tag I tag a content item to 'Education > Schools > Maths' ...but Maths is not shown to users, the content would appear on the frontend tagged to Schools instead. Taxonomy pages already roll-up tagging in this way.

I'm inclined to option 1 to keep things simple for publishers. There may a case for slimming down the taxonomy considerably if we're not using the lower levels.